### ISSUE TITLE
How much GPU memory we required for incremental training (10+10)

### ISSUE THREAD

**Issue by niranjansuthar70** [kshmelkov/incremental_detectors#14]
@kshmelkov Sir, I have trained the network for first 3 classes (GPU 1060 6GB) successfully, but when I am doing the incremental learning with 2 new classes (on GPU 1070 8 GB).
I am doing in this way, first train for 3 classes only (created separated dataset for it from voc07) and then tried to incremental training for 2 new classes (created dataset for only 2 classes).
During 2nd step, incremental training, I am getting error like this:

"resourceExhaustedError (see above for traceback): OOM when allocating tensor with shape"

I have used, reducing learning rate, reducing batch size but nothing working.
Can you please tell me the possible solutions for the above error or is it related to GPU memory requirements.

thanks.

Although I have made few changes in (distillation loss for xe and bbox) network.py (attaching file here)

  def compute_distillation_bbox_loss(self):
        #cached_classes = self.num_classes - self.subnet.num_classes
        bboxes_full = tf.reshape(self.bboxes_for_distillation, (-1, self.num_classes+1, 4))
        bboxes = bboxes_full[:, 1:self.subnet.num_classes+1, :]
        #print("Hello:1",bboxes.shape)
        cache_bboxes = tf.reshape(self.distillated_bboxes, (-1, self.subnet.num_classes+1, 4))[:, 1:, :]
        #print("Hello:2",cache_bboxes.shape)   

def compute_distillation_crossentropy_loss(self):
        #cached_classes = self.num_classes - self.subnet.num_classes
        logits = self.logits_for_distillation[:, :self.subnet.num_classes+1]
        #print(logits.shape)
